CDP4Composition.Tests/IconCacheService/IconCacheServiceIconCachServiceTestFixture.cs

```diff
@@ -61,5 +61,34 @@ public void VerityThatSameBitmapSourceIsReturned()
 
             Assert.AreSame(firstBitmapSource, secondBitmapSource);
         }
+
+
+        [Test]
+        public void VerityThatOverlayIconGetCached()
+        {
+            var stopWatchFirstFetch = new System.Diagnostics.Stopwatch();
+            var stopWatchSecondFetch = new System.Diagnostics.Stopwatch();
+            const string naturalLanguageIcon = "pack://application:,,,/CDP4Composition;component/Resources/Images/Thing/naturallanguage.png";
+            const string relationshipOverlayUri = "pack://application:,,,/CDP4Composition;component/Resources/Images/Log/linkgreen_16x16.png";
+            
+            var imageUri = new Uri(naturalLanguageIcon);
+            var overlayUri = new Uri(relationshipOverlayUri);
+
+            stopWatchFirstFetch.Start();
+            var firstBitmapSource = this.iconCacheService.QueryOverlayBitmapSource(imageUri, overlayUri);
+            stopWatchFirstFetch.Stop();
+
+            Assert.IsInstanceOf<BitmapSource>(firstBitmapSource);
+
+            stopWatchSecondFetch.Start();
+            var secondBitmapSource = this.iconCacheService.QueryOverlayBitmapSource(imageUri, overlayUri);
+            Assert.IsInstanceOf<BitmapSource>(secondBitmapSource);
+            stopWatchSecondFetch.Stop();
+
+            Assert.AreSame(firstBitmapSource, secondBitmapSource);
+
+            Assert.Greater(stopWatchFirstFetch.ElapsedMilliseconds, stopWatchSecondFetch.ElapsedMilliseconds);
```

Comment:
When for some reason (PC is busy) the second time DOES take longer to execute, you get a false negative.
The Assert.AreSame(firstBitmapSource, secondBitmapSource) already makes sure the same object is returned.

Reply:
Indeed, here maybe better to assert that a certain method in `iconCacheService` is called?